### PR TITLE
DOCSUP-14539: Edit and translate to Russian (MaterializedPostgreSQL settings)

### PR DESCRIPTION
**Pull request by olgarev**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Documentation for #28614, #28556

### CODE REVIEW

File: docs/ru/engines/database-engines/materialized-postgresql.md [hunk 1]
```diff
@@ -82,3 +86,10 @@ ENGINE = MaterializedPostgreSQL('postgres1:5432', 'postgres_database', 'postgres
 
 SELECT * FROM postgresql_db.postgres_table;
 ```
+## Примечания {#notes}
+
+### Сбой слота логической репликации {logical-replication-slot-failover}
+
+Слоты логической репликации, которые есть на основном сервере, не доступны на резервных репликах.
+Поэтому в случае сбоя новый основной сервер (который раньше был резевным физическим сервером) не будет знать о слотах, которые были расположены на вышедшем из строя основном сервере. Это приведет к нарушению репликации из PostgreSQL.
```
Comment (kssenii) [inline docs/ru/engines/database-engines/materialized-postgresql.md:94]:
```suggestion
Поэтому в случае сбоя новый основной сервер (который раньше был резевным) не будет знать о слотах репликации, которые были созданы на вышедшем из строя основном сервере. Это приведет к нарушению репликации из PostgreSQL.
```

Comment (kssenii) [reply docs/ru/engines/database-engines/materialized-postgresql.md:94]: also may be add this example https://github.com/ClickHouse/ClickHouse/issues/28529#issuecomment-927209693 (with addition of  the author nickname)

**Review by kssenii — APPROVED** [review]

File: docs/en/engines/database-engines/materialized-postgresql.md [hunk 2]
```diff
@@ -109,8 +109,54 @@ SELECT * FROM postgresql_db.postgres_table;
 
 ## Notes {#notes}
 
-- Failover of the logical replication slot.
+### Failover of the logical replication slot {logical-replication-slot-failover}
```
Comment (gyuton) [inline docs/en/engines/database-engines/materialized-postgresql.md:112]:
```suggestion
### Failover of the logical replication slot {#logical-replication-slot-failover}
```

File: docs/ru/engines/database-engines/materialized-postgresql.md [hunk 3]
```diff
@@ -82,3 +86,56 @@ ENGINE = MaterializedPostgreSQL('postgres1:5432', 'postgres_database', 'postgres
 
 SELECT * FROM postgresql_db.postgres_table;
 ```
+## Примечания {#notes}
+
+### Сбой слота логической репликации {logical-replication-slot-failover}
```
Comment (gyuton) [inline docs/ru/engines/database-engines/materialized-postgresql.md:91]:
```suggestion
### Сбой слота логической репликации {#logical-replication-slot-failover}
```

File: docs/ru/engines/database-engines/materialized-postgresql.md [hunk 4]
```diff
@@ -82,3 +86,56 @@ ENGINE = MaterializedPostgreSQL('postgres1:5432', 'postgres_database', 'postgres
...
+
+Слоты логической репликации, которые есть на основном сервере, не доступны на резервных репликах.
+Поэтому в случае сбоя новый основной сервер (который раньше был резевным) не будет знать о слотах репликации, которые были созданы на вышедшем из строя основном сервере. Это приведет к нарушению репликации из PostgreSQL.
+Решением этой проблемы может стать ручное управление слотами репликации и определение постоянного слота репликации (об этом можно прочитать [здесь](https://patroni.readthedocs.io/en/latest/SETTINGS.html)). Этот слот нужно назначить с помощью настройки `materialized_postgresql_replication_slot`, и он должен быть экспортирован в параметре `EXPORT SNAPSHOT`. Идентификатор снэпшота нужно передать в настройке `materialized_postgresql_snapshot`.
```
Comment (gyuton) [inline docs/ru/engines/database-engines/materialized-postgresql.md:95]:
```suggestion
Решением этой проблемы может стать ручное управление слотами репликации и определение постоянного слота репликации (об этом можно прочитать [здесь](https://patroni.readthedocs.io/en/latest/SETTINGS.html)). Этот слот нужно назначить с помощью настройки [materialized_postgresql_replication_slot](../../operations/settings/settings.md#materialized-postgresql-replication-slot), и он должен быть экспортирован в параметре `EXPORT SNAPSHOT`. Идентификатор снэпшота нужно передать в настройке [materialized_postgresql_snapshot](../../operations/settings/settings.md#materialized-postgresql-snapshot).
```

File: docs/en/engines/database-engines/materialized-postgresql.md [hunk 5]
```diff
@@ -109,8 +109,54 @@ SELECT * FROM postgresql_db.postgres_table;
...
 
 Logical Replication Slots which exist on the primary are not available on standby replicas.
 So if there is a failover, new primary (the old physical standby) won’t be aware of any slots which were existing with old primary. This will lead to a broken replication from PostgreSQL.
 A solution to this is to manage replication slots yourself and define a permanent replication slot (some information can be found [here](https://patroni.readthedocs.io/en/latest/SETTINGS.html)). You'll need to pass slot name via `materialized_postgresql_replication_slot` setting, and it has to be exported with `EXPORT SNAPSHOT` option. The snapshot identifier needs to be passed via `materialized_postgresql_snapshot` setting.
```
Comment (gyuton) [inline docs/en/engines/database-engines/materialized-postgresql.md:116]:
```suggestion
A solution to this is to manage replication slots yourself and define a permanent replication slot (some information can be found [here](https://patroni.readthedocs.io/en/latest/SETTINGS.html)). You'll need to pass slot name via [materialized_postgresql_replication_slot](../../operations/settings/settings.md#materialized-postgresql-replication-slot) setting, and it has to be exported with `EXPORT SNAPSHOT` option. The snapshot identifier needs to be passed via [materialized_postgresql_snapshot](../../operations/settings/settings.md#materialized-postgresql-snapshot) setting.
```

File: docs/ru/engines/database-engines/materialized-postgresql.md [hunk 6]
```diff
@@ -82,3 +86,56 @@ ENGINE = MaterializedPostgreSQL('postgres1:5432', 'postgres_database', 'postgres
...
+```sql
+CREATE DATABASE demodb
+ENGINE = MaterializedPostgreSQL('postgres1:5432', 'postgres_database', 'postgres_user', 'postgres_password')
+SETTINGS
+  materialized_postgresql_replication_slot = 'clickhouse_sync',
+  materialized_postgresql_snapshot = '0000000A-0000023F-3',
+  materialized_postgresql_tables_list = 'table1,table2,table3';
+```
+
+4. Когда БД в ClickHouse создастся, завершите транзакцию в PostgreSQL. Убедитесь, что репликация продолжается после сбоя:
```
Comment (olgarev) [inline docs/ru/engines/database-engines/materialized-postgresql.md:137]:
```suggestion
4. Когда начнет выполняться репликация БД в ClickHouse, прервите транзакцию в PostgreSQL. Убедитесь, что репликация продолжается после сбоя:
```

File: docs/ru/engines/database-engines/materialized-postgresql.md (hunk 4 above)
Comment (olgarev) [inline docs/ru/engines/database-engines/materialized-postgresql.md]:
```suggestion
Решением этой проблемы может стать ручное управление слотами репликации и определение постоянного слота репликации (об этом можно прочитать [здесь](https://patroni.readthedocs.io/en/latest/SETTINGS.html)). Этот слот нужно передать с помощью настройки `materialized_postgresql_replication_slot`, и он должен быть экспортирован в параметре `EXPORT SNAPSHOT`. Идентификатор снэпшота нужно передать в настройке `materialized_postgresql_snapshot`.
```

File: docs/ru/engines/database-engines/materialized-postgresql.md [hunk 7]
```diff
@@ -82,3 +86,56 @@ ENGINE = MaterializedPostgreSQL('postgres1:5432', 'postgres_database', 'postgres
 
 SELECT * FROM postgresql_db.postgres_table;
 ```
+## Примечания {#notes}
+
+### Сбой слота логической репликации {#logical-replication-slot-failover}
+
+Слоты логической репликации, которые есть на основном сервере, не доступны на резервных репликах.
+Поэтому в случае сбоя новый основной сервер (который раньше был резевным) не будет знать о слотах репликации, которые были созданы на вышедшем из строя основном сервере. Это приведет к нарушению репликации из PostgreSQL.
+Решением этой проблемы может стать ручное управление слотами репликации и определение постоянного слота репликации (об этом можно прочитать [здесь](https://patroni.readthedocs.io/en/latest/SETTINGS.html)). Этот слот нужно назначить с помощью настройки [materialized_postgresql_replication_slot](../../operations/settings/settings.md#materialized-postgresql-replication-slot), и он должен быть экспортирован в параметре `EXPORT SNAPSHOT`. Идентификатор снэпшота нужно передать в настройке [materialized_postgresql_snapshot](../../operations/settings/settings.md#materialized-postgresql-snapshot).
```
Comment (olgarev) [inline docs/ru/engines/database-engines/materialized-postgresql.md:95]:
```suggestion
Решением этой проблемы может стать ручное управление слотами репликации и определение постоянного слота репликации (об этом можно прочитать [здесь](https://patroni.readthedocs.io/en/latest/SETTINGS.html)). Этот слот нужно назначить с помощью настройки [materialized_postgresql_replication_slot](../../operations/settings/settings.md#materialized-postgresql-replication-slot), и он должен быть экспортирован в параметре `EXPORT SNAPSHOT`. Идентификатор снэпшота нужно передать в настройке [materialized_postgresql_snapshot](../../operations/settings/settings.md#materialized-postgresql-snapshot).
```

**Comment by olgarev** [comment]
@kssenii Hi, the PR is ready now, please review it.

File: docs/en/engines/database-engines/materialized-postgresql.md [hunk 8]
```diff
@@ -109,8 +109,54 @@ SELECT * FROM postgresql_db.postgres_table;
...
-A solution to this is to manage replication slots yourself and define a permanent replication slot (some information can be found [here](https://patroni.readthedocs.io/en/latest/SETTINGS.html)). You'll need to pass slot name via `materialized_postgresql_replication_slot` setting, and it has to be exported with `EXPORT SNAPSHOT` option. The snapshot identifier needs to be passed via `materialized_postgresql_snapshot` setting.
+A solution to this is to manage replication slots yourself and define a permanent replication slot (some information can be found [here](https://patroni.readthedocs.io/en/latest/SETTINGS.html)). You'll need to pass slot name via [materialized_postgresql_replication_slot](../../operations/settings/settings.md#materialized-postgresql-replication-slot) setting, and it has to be exported with `EXPORT SNAPSHOT` option. The snapshot identifier needs to be passed via [materialized_postgresql_snapshot](../../operations/settings/settings.md#materialized-postgresql-snapshot) setting.
+
+**Example (from [@bchrobot](https://github.com/bchrobot))** 
+
+1. Configure replication slot in PostgreSQL:
```
Comment (kssenii) [inline docs/en/engines/database-engines/materialized-postgresql.md:120]: Can you please add a note that this should be only used only if it is actually needed and if there is no real need for that or full understanding why, then this setting should not be used.

Comment (olgarev) [inline docs/en/engines/database-engines/materialized-postgresql.md:120]:
```suggestion
1. Configure replication slot in PostgreSQL. Please note that this should be used only if it is actually needed. If there is no real need for that or full understanding why, then replication slot should not be used.
```

File: docs/en/engines/database-engines/materialized-postgresql.md [hunk 9]
```diff
@@ -109,8 +109,54 @@ SELECT * FROM postgresql_db.postgres_table;
...
+        type: logical
+        database: demodb
+        plugin: pgoutput
+```
+
+2. Wait for replication slot to be ready, then begin a transaction and export the transaction snapshot identifier:
+
+```sql
+begin;
+select pg_export_snapshot();
```
Comment (olgarev) [inline docs/en/engines/database-engines/materialized-postgresql.md:144]:
```suggestion
BEGIN;
SELECT pg_export_snapshot();
```

File: docs/ru/engines/database-engines/materialized-postgresql.md [hunk 10]
```diff
@@ -82,3 +86,56 @@ ENGINE = MaterializedPostgreSQL('postgres1:5432', 'postgres_database', 'postgres
...
+
+### Сбой слота логической репликации {#logical-replication-slot-failover}
+
+Слоты логической репликации, которые есть на основном сервере, не доступны на резервных репликах.
+Поэтому в случае сбоя новый основной сервер (который раньше был резевным) не будет знать о слотах репликации, которые были созданы на вышедшем из строя основном сервере. Это приведет к нарушению репликации из PostgreSQL.
+Решением этой проблемы может стать ручное управление слотами репликации и определение постоянного слота репликации (об этом можно прочитать [здесь](https://patroni.readthedocs.io/en/latest/SETTINGS.html)). Этот слот нужно передать с помощью настройки [materialized_postgresql_replication_slot](../../operations/settings/settings.md#materialized-postgresql-replication-slot), и он должен быть экспортирован в параметре `EXPORT SNAPSHOT`. Идентификатор снэпшота нужно передать в настройке [materialized_postgresql_snapshot](../../operations/settings/settings.md#materialized-postgresql-snapshot).
+
+**Пример (от [@bchrobot](https://github.com/bchrobot))** 
+
+1. Сконфигурируйте слот репликации в PostgreSQL:
```
Comment (olgarev) [inline docs/ru/engines/database-engines/materialized-postgresql.md:99]:
```suggestion
1. Сконфигурируйте слот репликации в PostgreSQL. Имейте в виду, что это стоит делать только если есть реальная необходимость. Если такой необходимости нет, или если нет полного понимания того, как это работает, слот репликации конфигурировать не стоит.
```

File: docs/ru/engines/database-engines/materialized-postgresql.md [hunk 11]
```diff
@@ -82,3 +86,56 @@ ENGINE = MaterializedPostgreSQL('postgres1:5432', 'postgres_database', 'postgres
...
+        type: logical
+        database: demodb
+        plugin: pgoutput
+```
+
+2. Дождитесь готовности слота репликации, затем инициируйте транзакцию и экспортируйте идентификатор снэпшота этой транзакции:
+
+```sql
+begin;
+select pg_export_snapshot();
```
Comment (olgarev) [inline docs/ru/engines/database-engines/materialized-postgresql.md:123]:
```suggestion
BEGIN;
SELECT pg_export_snapshot();
```

**Pull request merged**